app/controllers/lettings_logs_controller.rb

```diff
@@ -115,14 +115,50 @@ def update_logs
     end
   end
 
+  def bulk_uploads
+    return render_not_authorized unless current_user.support?
+
+    @filter_type = "lettings_bulk_uploads"
+
+    if params[:organisation_id].present? && params[:clear_old_filters].present?
+      redirect_to clear_filters_path(filter_type: @filter_type, organisation_id: params[:organisation_id]) and return
+    end
+
+    uploads = BulkUpload.lettings.where("created_at >= ?", 30.days.ago)
+    unpaginated_filtered_uploads = filter_manager.filtered_uploads(uploads, search_term, filter_manager.session_filters)
+
+    @pagy, @bulk_uploads = pagy(unpaginated_filtered_uploads)
+    @search_term = search_term
+    @total_count = uploads.size
+    @searched = search_term.presence
+    render "bulk_upload_shared/uploads"
+  end
+
+  def download_bulk_upload
+    bulk_upload = BulkUpload.find(params[:id])
+    downloader = BulkUpload::Downloader.new(bulk_upload:)
+
+    if Rails.env.development?
+      downloader.call
+      send_file downloader.path, filename: bulk_upload.filename, type: "text/csv"
+    else
+      presigned_url = downloader.presigned_url
+      redirect_to presigned_url, allow_other_host: true
+    end
+  end
```

Comment:
What if I go to `https://review.submit-social-housing-data.communities.gov.uk/2666/lettings-logs/bulk-uploads/{id}/download` while signed in as non-support user?

Reply:
Nice catch, that is possible, I will prevent it.